bcgov/mds
# Feature/mds 3238 - Populate conditions with variables

# Main

- Ability to add variables to the permit conditions on the UI
- Ability to map those variables to data points on the NoW, permit, mine, etc and display on the permit pdf in place of the variables
- The "Permit Condition Management" view displays a default list of reclamation activities, The draft permit tab displays only the activities related to the NoW.

# Other
- Updated test coverage on all condition components. 
- Added a modal to view all permit conditions on the Permit table. 
![Screen Shot 2021-10-15 at 5 04 02 PM](https://user-images.githubusercontent.com/25125247/137565762-a3c37b4e-fae1-4e7f-a71a-0af4ac74995b.png)
![Screen Shot 2021-10-15 at 5 04 49 PM](https://user-images.githubusercontent.com/25125247/137565793-6fcbad1f-9a50-45ea-aecd-6b7cd92debe8.png)
![Screen Shot 2021-10-15 at 5 14 02 PM](https://user-images.githubusercontent.com/25125247/137566151-3f6ae7fa-67ad-4839-8c2d-5038c9d391b5.png)




# How to test

- Create a draft permit -> alter the conditions to include some or all of the variables listed in the menu -> Click "download Draft" and confirm that the pdf is now displaying the correct data in place of the variable names. 
Test Cases 
1. when you click the edit pencil, a menu appears in the top left corner. 
2. Selecting an option from the menu adds a variable to the input box in this format: {variable}
3. When you click save on the permit condition, the variable gets highlighted yellow. 
4. When you click "download Draft Permit" it downloads successfully, the variable is either replaced with the data or an empty string if the data is null. 

Repeat steps 1-3 on the "Permit Condition Management" tab under the Admin menu.

# Notes
![Screen Shot 2021-10-15 at 1 51 01 PM](https://user-images.githubusercontent.com/25125247/137551901-f7f2c4b3-43b7-4dee-8d75-136023352930.png)
![Screen Shot 2021-10-15 at 1 51 20 PM](https://user-images.githubusercontent.com/25125247/137551937-74d023cf-7c0a-4d1a-92c3-0f2cbdd96bbe.png)
![Screen Shot 2021-10-15 at 1 51 34 PM](https://user-images.githubusercontent.com/25125247/137551966-26da0c24-ce38-41bb-8ea8-16918b1ac960.png)
![Screen Shot 2021-10-15 at 1 51 53 PM](https://user-images.githubusercontent.com/25125247/137552006-a1b5fb3e-7008-4e93-96f1-fa636e4a09fe.png)
![Screen Shot 2021-10-15 at 1 52 31 PM](https://user-images.githubusercontent.com/25125247/137552072-80eb22a1-5d1c-43c8-b47c-96337dcb7f6f.png)
![Screen Shot 2021-10-15 at 1 53 01 PM](https://user-images.githubusercontent.com/25125247/137552139-6bcf9c6e-7efa-4014-9125-db44cc4091df.png)
![Screen Shot 2021-10-15 at 1 53 33 PM](https://user-images.githubusercontent.com/25125247/137552211-783e48a0-754c-4c0f-9230-8830c13ee316.png)
